### PR TITLE
Filter private addresses from gossip

### LINKED ISSUE #799: Filter addresses that are not reachable
something to add ASAP. It's important to filter addresses that are not reachable `10.0.0.0/8` or `0.0.0.0`.

### PR DESCRIPTION
**Pull request by mzabaluev**
Resolves #799

### CODE REVIEW

**Comment by mzabaluev** [comment]
This results in log spam if the connecting node's reported address in gossip is not acceptable.
Should this be fixed by:
1. disconnecting such nodes completely rather than just filtering out bad gossip items;
2. removing the log message, to detriment of the network admin getting a clue about the configuration problem?

File: jormungandr/src/settings/start/config.rs [hunk 1]
```diff
@@ -79,6 +79,10 @@ pub struct P2pConfig {
     /// help the different modules of the P2P topology engine to determine the
     /// best possible neighborhood.
     pub topics_of_interest: Option<BTreeMap<Topic, InterestLevel>>,
+
+    /// Whether to allow non-public IP addresses on the network
+    #[serde(default)]
+    pub allow_private_addresses: bool,
```
Comment (NicolasDP) [inline jormungandr/src/settings/start/config.rs]: what is the default value? It's not obvious to me

could you add it in the documentation too? I'd like to see the default set to _false_ by default. Is it the case?

File: jormungandr/src/network/subscription.rs [hunk 2]
```diff
@@ -105,12 +107,40 @@ where
...
+        node.is_global()
+    };
+    if !accepted {
+        if let Some(addr) = node.address() {
+            warn!(
+                logger,
+                "node {} has unacceptable IP address {}, pruned from gossip",
+                node.id(),
+                addr.ip()
+            );
```
Comment (NicolasDP) [inline jormungandr/src/network/subscription.rs]: We will get a lot of logs, could we filter them all and then says something like:

filtering out these `[addresses]` from the gossips from `address` of the one who sent us these gossips?

File: jormungandr/src/settings/start/config.rs (hunk 1 above)
Comment (mzabaluev) [reply jormungandr/src/settings/start/config.rs]: Yes, it should be false by default, but I don't know how to specify this explicitly for serde-yaml. I'll add the information to the docs.

Comment (NicolasDP) [reply jormungandr/src/settings/start/config.rs]: no worries. Just needs it to be documented in the comment message

File: jormungandr/src/network/subscription.rs (hunk 2 above)
Comment (mzabaluev) [reply jormungandr/src/network/subscription.rs]: Done in a follow-up comment. The peer address should be already added to the `Logger` context of the connection. If not, it should be fixed elsewhere.

**Pull request merged**